.pre-commit-config.yaml

```diff
@@ -41,3 +41,39 @@ repos:
         # needed to make excludes in pyproject.toml work
         # see here https://github.com/econchick/interrogate/issues/60#issuecomment-735436566
         pass_filenames: false
+  - repo: local
```

Comment:
this is probably more of a philosophical question for you (@ericmjl ) and @drbenvincent  

initially this was just checking spelling outside of the jupyter notebooks, which to me definitely feels like a `pre-commit` check, but now the check is including the notebook to markdown conversion to find spelling mistakes in these notebooks.  maybe this highlights a bit of scope creep for this one PR / issue because purely as a `pre-commit` check I think it makes sense to just look at the `.py` and `.md` files and then the notebook spelling check is more of a CI check

so, it's up to you two but i'm happy to keep plugging away at this PR with the updated doc changes and rationale changes but maybe it makes sense to prune back this PR to just the base `codespell` checks then cherry-pick out the commits to a new PR for an actual CI check for jupyter notebooks since those should be more static and don't need to be checked on every commit -- thoughts?

Reply:
Hi @westford14. Yeah, I was thinking the same thing - happy for you to do this.